python/fledge/common/utils.py

```diff
@@ -85,3 +85,42 @@ def decorator(Class):
 def eprint(*args, **kwargs):
     """ eprintf -- convenience print function that prints to stderr """
     print(*args, *kwargs, file=sys.stderr)
+
+
+def read_os_release():
+    """ General information to identifying the operating system """
+    import ast
+    import re
+    os_details = {}
+    try:
+        filename = '/etc/os-release'
+        f = open(filename)
+    except FileNotFoundError:
+        filename = '/usr/lib/os-release'
+        f = open(filename)
+
+    for line_number, line in enumerate(f, start=1):
+        line = line.rstrip()
+        if not line or line.startswith('#'):
+            continue
+        m = re.match(r'([A-Z][A-Z_0-9]+)=(.*)', line)
+        if m:
+            name, val = m.groups()
+            if val and val[0] in '"\'':
+                val = ast.literal_eval(val)
+            os_details.update({name: val})
+    return os_details
+
+
+def is_debian():
+    """
+        To check if the Operating system is of Debian type or Not
```

Comment:
of Debian type or Not
Debian based or Not

Actually we are checking if Redhat family or not

Reply:
is_redhat_based/is_redhat_family will keep it more cleaner along with existing changed for yum vs apt.

e.g. 
`pkg_mgt = 'yum' if is_redhat_based() else 'apt'`